masu/processor/tasks.py

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Celery task definitions."""
+
+# pylint: disable=unused-argument, fixme
+# FIXME: temporary module-wide disable until tasks are fully implemented.
+
+from unittest import mock
+
+from celery.utils.log import get_task_logger
+
+from .celery.worker import QUEUE
+from .exceptions import MasuProcessingError
+
+
+LOG = get_task_logger(__name__)
+
+
+@QUEUE.task(autoretry_for=(MasuProcessingError,),
+            retry_backoff=True)
+def check_update(**kwargs):
+    """Task that checks SNS and/or S3 for updated CUR files."""
+    # TODO: 1. look for new data.
+    # TODO: 2. when Step 1 is True, add a cache_from_s3() task to the queue.
+    return True
+
+
+@QUEUE.task(autoretry_for=(MasuProcessingError,),
+            retry_backoff=True)
+def cache_from_s3(**kwargs):
+    """
+    Task that downloads CUR from S3.
+
+    The downloaded files are stored in a specified directory for processing.
+    """
+    # TODO: 1. download new content (compare E-Tags)
+    # TODO: 2. when download completes, add a process_cost_usage_report() task
+    # TODO:    to the queue.
+    pass
+
+
+@QUEUE.task(autoretry_for=(MasuProcessingError,),
+            retry_backoff=True)
+def process_cost_usage_report(**kwargs):
+    """Task that processes CUR files, inserting data into the Koku DB."""
+    # TODO: 1. Unzip and process each .csv.gz file in processing directory.
+    # TODO: 2. Insert any new data found in Step 1, skipping existing data.
+    pass
+
+
+@QUEUE.task(track_started=True)
+def test(args):
+    """Test task. Used only in testing."""
+    LOG.debug(args)
+    return mock.Mock(name=__name__, args=args)
```

Comment:
I'm not sure if we should include test libraries in the production code.  Maybe we can do something like:
```
print("Start Task args: {}".format(args))
time.sleep(10)
print("End Task") instead?
```

Reply:
I fully expect this method to get removed "Very Soon" once we have something more useful to do.